pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,390 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from time import sleep
+from time import time as now
+
+from io import BytesIO
+
+import numpy as np
+from pyvisa import VisaIOError
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, truncated_range
+
+
+class HP8753E(Instrument):
+    """Represents the HP8753E Vector Network Analyzer
+    and provides a high-level interface for taking scans of the
+    scattering parameters.
+
+    Keyword arguements:
+    adapter -- <placeholder for description of pymeasure.adapter>
+    name -- <str> describing the instrument
+    """
+
+    def __init__(
+        self,
+        adapter=None,
+        name=None,
```

Comment:
Normally, the default name is the model of the instrument, in this case for example "HP8753E VNA" or just "HP8753E", but I see your idea here.

The idea is, that a user can give a name to all his instruments, even though they might be of the same model (therefore instruments must have a "name" parameter).

If you wish to know the serial number etc. you can always use the `id` method.

Therefore the `name` is user defined, while the `id` is device defined.

Reply:
I can remove the extra fluff of querying the device to name it if it is desired for consistency. Let me know if you want to keep it or toss it.

I'm inclined to keep it unless it would cause something to break by having two devices with the same name or something.